### ISSUE TITLE
[ION-1354] Feat: Windows support

### ISSUE THREAD

**Issue by jayair** [sst/sst#4685]
Will work on this once we round out most of the components.

<sub>[ION-1354](https://linear.app/sst-dev/issue/ION-1354/feat-windows-support)</sub>

**Comment by nacholupotti** [comment]
+1 to get windows support. Looking forward to see this! Thanks for all your work

**Comment by JasonWeise** [comment]
I hate to be "that guy" but like many others I have been waiting for Windows support since v3 release and would love to know if there is a rough ETA. I know it is getting asked quite a bit, but our dev team is pretty much all Windows and I would rather not start a new project in v2 if v3 Windows support is very near.

With all respect, I totally appreciate wanting to get all the other moving pieces right first as jayair has said a few times now, but as it (v3) was released officially a while back now, I would have hoped we would have been at the point of having Windows support by now.

**Comment by sam-masscreations** [comment]
+1, Any update on Windows support release?

**Comment by jayair** [comment]
Should be in the next month or so.

**Comment by mph6083** [comment]
Is there any update on Windows support?

**Comment by mmixan** [comment]
Any update here?  Want to move back to SST land, but would prefer to start with v3 rather than starting with V2 and having to upgrade to V3 once the Windows CLI is available.

**Comment by goto3** [comment]
A LOT of people are waiting for this, me personally since may 2024... WSL did not make it for me, it is just too buggy, slow and had issues with git, eslint, etc

**Comment by Jakub-Izbicki** [comment]
Hello @jayair 
Is there any new ETA for windows support?

**Comment by gabriel-garcia-prado** [comment]
Hello @jayair
Is there any new ETA for windows support?

**Comment by jayair** [comment]
Working on it in February, fingers crossed!

**Comment by rkarbowiak** [comment]
Is there any progress @jayair? Rooting for this one!

**Comment by zcoffin86** [comment]
Squeaky wheel here. 10 months after this issue opened, any updates on ETA?

**Comment by jdanl89** [comment]
@jayair how can I and the community be of assistance to get this off the ground?

**Comment by TeamVASH** [comment]
I want to move to production soon for my SST next.js site, but I'm stuck on v2, waiting for windows support too!

**Comment by markoj3s** [comment]
@jayair Any update?

**Comment by jsadoski-rockhall** [comment]
Sorry to add to the bumps, but I am looking forward to this feature.

I will add however that 2 of 3 on my team did end up using WSL and upgraded to v3. The third is on Mac, so no issues there. Aside from some small WSL configuration pains, things have been great. This is no longer a blocker for us.

I also briefly tried GitHub Codespaces and Visual Studio Code Dev Containers. Both were great as well. I settled on WSL because it provided the best access to compute while developing.

While I do think this is a necessary feature, at this point it's more of a convenience for our team. Thank you!

**Comment by sam-masscreations** [comment]
@jayair Hows the windows support coming along?

**Comment by jayair** [comment]
We are working on it!

**Comment by rcoundon** [comment]
We're pretty desperate for this but I was wondering, is there any reason that running a Docker container with something like this wouldn't work or be recommended:

```Dockerfile
# Use Node.js LTS as the base image
FROM node:22-slim

# Set working directory
WORKDIR /app

# Install system dependencies
RUN apt-get update && apt-get install -y \
    git \
    curl \
    python3 \
    build-essential \
    && rm -rf /var/lib/apt/lists/*

# Install SST v3 globally
RUN npm install -g sst@3

# Create startup script with logging
RUN echo '#!/bin/bash\nset -x\necho "Starting SST container..."\necho "Arguments: $@"\necho "Current directory: $(pwd)"\necho "Directory contents:"\nls -la\nsst "$@" || { echo "SST command failed with exit code $?"; exit 1; }' > /docker-entrypoint.sh \
    && chmod +x /docker-entrypoint.sh

# Set entrypoint to our debug script
ENTRYPOINT ["/docker-entrypoint.sh"]
```

**Comment by jayair** [comment]
@jsadoski-rockhall any chance you can share what your team did?

I think @rcoundon will probably find it helpful.

**Comment by jdanl89** [comment]
@rcoundon there are options that Microsoft provides that could prove useful too. WSL and VS Code's Remote Development extension. ([Link](https://code.visualstudio.com/docs/remote/remote-overview)) WSL allows you to run Linux in parallel to your windows OS with a shared file system. The Remote Development extension allows you to SSH into a Linux box from your Windows machine.

I use the VS Code extension for personal projects. I have an old laptop that I installed Ubuntu on, and I SSH into that to work on projects that I can't do on Windows.

**Comment by rcoundon** [comment]
> [@rcoundon](https://github.com/rcoundon) there are options that Microsoft provides that could prove useful too. WSL and VS Code's Remote Development extension. ([Link](https://code.visualstudio.com/docs/remote/remote-overview)) WSL allows you to run Linux in parallel to your windows OS with a shared file system. The Remote Development extension allows you to SSH into a Linux box from your Windows machine.
> 
> I use the VS Code extension for personal projects. I have an old laptop that I installed Ubuntu on, and I SSH into that to work on projects that I can't do on Windows.

Thanks - yeah, we've been trying to get WSL working without much luck but may be making progress now. I think the breakthrough was avoiding using files on the Windows file system via a mount and instead using files downloaded from Github directly to the Linux filesystem.

I'm writing up some notes as I go so I'l share somewhere appropriate if I can get it working well

**Comment by mph6083** [comment]
@rcoundon We were having similar problems. What worked for our team was using VS code's remote explorer and fully using the linux filesystem, installing a separate node/npm on WSL, and ensuring the commands were coming from the linux version of node.

**Comment by rcoundon** [comment]
> @rcoundon We were having similar problems. What worked for our team was using VS code's remote explorer and fully using the linux filesystem, installing a separate node/npm on WSL, and ensuring the commands were coming from the linux version of node.

Yes, that's the conclusion I'm reaching too. Thank you

**Comment by aphex** [comment]
@mph6083 Definitely had the same experience for SST v2 as well. Keeping everything in Linux worked great, however our project uses Electron applications and trying to juggle SST in Linux and with windows applications was a nightmare. 

However rare technical reasons aside we really do need proper windows support here. 🙏

**Comment by pstovik** [comment]
Hi @jayair, is there any ETA / Roadmap / release plan when Windows support should be ready? We have SST2 => SST3 migration in progress so whether to teach all the Windows developers how to configure and work with WSL ... or wait some days (and use our project as beta testers) ? :-)

**Comment by jayair** [comment]
We do have an initial version some folks are testing right now. You can try it with `sst@0.0.0-1743715512`. The multiplexer doesn't work yet. So you'll need to run `sst dev --mode=mono`.

**Comment by rcoundon** [comment]
> We do have an initial version some folks are testing right now. You can try it with `sst@0.0.0-1743715512`. The multiplexer doesn't work yet. So you'll need to run `sst dev --mode=mono`.

Hi Jay - is `--mode=mono` different to `mode=basic` ?

**Comment by pstovik** [comment]
@jayair , **thanks for sharing**, I was curious so I did a test on our project via Windows CLI (in WSL the project is working great)
I know it is still pre-alfa so not blaming anyhow, just sharing my "test results" in case somebody would be interested 🙂

our project context: pnpm monorepo with workspaces, AWS typescript lambdas, dynamoDB + SQS, react / vite build

1) generally working installation (but slow, probably also due antivirus / filesystem checks present on Windows OS)

2) failures when trying to run `sst diff` 
> Error: Could not resolve "proxy-from-env" packages/be-main/node_modules/@vega/be-main-auth/node_modules/axios/lib/adapters/http.js:7:25
> Could not resolve "follow-redirects" packages/be-main/node_modules/@vega/be-main-auth/node_modules/axios/lib/adapters/http.js:11:28
> Cannot read file "packages/be-admin/node_modules/@vega/utils": Incorrect function. packages/be-admin/src/taskQueue/adminTaskQueueHandler.ts:5:28

3) weird / completely different experience when running `sst dev --mode=mono`
- only "Waiting for invocations..." message in console

4) no meaningful output for `sst dev`

5) running `sst shell vite` seems to not have proper binding / linking to the resources

**Comment by thdxr** [comment]
there are some updates to in sst@0.0.0-1745853910 

@pstovik i actually wasn't able to recreate most of your issues but can you try the latest snapshot?

**Comment by jayair** [comment]
With [v3.14](https://github.com/sst/sst/releases/tag/v3.14.0) native Windows support is now in beta.

Please upgrade and give it a try. If you run into any problems, open a new issue for that specific problem.

I'm going to close this issue.

**Comment by pstovik** [comment]
FYI: I have just run the Windows CLI (sst@3.14.14) and at least the errors I mentioned as number 2 in my previous post are still there

in this case I run "sst dev" and as is gets to building of lambda handler code it logs error like following

```
[Function] |  Build Error packages/be-main-auth/src/auth/authHandler.authHandler
[Function] |  ↳ Could not resolve "proxy-from-env" packages/be-main-auth/node_modules/axios/lib/adapters/http.js:7:25
[Function] |  ↳ Could not resolve "follow-redirects" packages/be-main-auth/node_modules/axios/lib/adapters/http.js:11:28
[Function] |  ↳ Could not resolve "form-data" packages/be-main-auth/node_modules/axios/lib/platform/node/classes/FormData.js:1:21
...
[Function] |  ↳ Could not resolve "aws4fetch" packages/be-main-auth/node_modules/sst/dist/aws/client.js:1:26
...
[Function] |  ↳ Cannot read file "packages/be-main-auth/node_modules/ulid": Incorrect function. packages/be-main-auth/src/auth/createTenantId.ts:1:21
```

worth mentioning - our project is pnpm with workspaces, different "workspace package" may implement zero or multiple SST/lambda handlers, "workspace package" has own package.json + tsconfig.json
